### PR TITLE
Adding sample weight and keras test to R-Square

### PR DESCRIPTION
**Pull request by SSaishruthi**
Added:
1. `Sample Weight` parameter
2. Keras test

@Squadrick Can you please take a look at it?

### CODE REVIEW

File: tensorflow_addons/metrics/r_square_test.py [hunk 1]
```diff
@@ -80,6 +82,23 @@ def test_r2_random_score(self):
...
+        model.add(layers.Dense(1))
+        # initialize metric
+        r = RSquare()
+        # compile the model
+        model.compile(loss="mean_squared_error",
+                      optimizer="adam",
+                      metrics=[r])
 
-if __name__ == '__main__':
-    tf.test.main()
```
Comment (seanpmorgan) [inline tensorflow_addons/metrics/r_square_test.py]: Why remove these two?

File: tensorflow_addons/metrics/r_square.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def __init__(self, name='r_square', dtype=tf.float32):
         self.res = self.add_weight("residual", initializer="zeros")
         self.count = self.add_weight("count", initializer="zeros")
 
-    def update_state(self, y_true, y_pred):
+    def update_state(self, y_true, y_pred, sample_weight=None):
```
Comment (seanpmorgan) [inline tensorflow_addons/metrics/r_square.py]: I believe `sample_weight` needs to be applied? See https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/metrics.py#L317-L329

Comment (SSaishruthi) [reply tensorflow_addons/metrics/r_square.py]: For now, we are adding it to be done. Will create a new PR with implementation for the same.

**Comment by gabrieldemarmiesse** [comment]
I'll close this pull request as it has been inactive for a while now. Feel free to comment if you want me to reopen the pull request.

**Pull request closed without merging**